datafusion/core/src/datasource/view.rs

```diff
@@ -342,6 +355,65 @@ mod tests {
         Ok(())
     }
 
+    #[tokio::test]
+    async fn query_view_with_filter_and_limit() -> Result<()> {
+        let session_ctx = SessionContext::with_config(
+            SessionConfig::new().with_information_schema(true),
+        );
+
+        session_ctx
+            .sql("CREATE TABLE abc AS VALUES (1,2,3), (4,5,6), (7,8,9)")
+            .await?
+            .collect()
+            .await?;
+
+        let view_sql = "CREATE VIEW xyz AS SELECT column1, column2 FROM abc";
+        session_ctx.sql(view_sql).await?.collect().await?;
+
+        let results = session_ctx.sql("SELECT * FROM information_schema.tables WHERE table_type='VIEW' AND table_name = 'xyz'").await?.collect().await?;
+        assert_eq!(results[0].num_rows(), 1);
+
+        let result_plan = session_ctx
+            .sql("EXPLAIN CREATE VIEW efg as SELECT column1 FROM xyz WHERE column2 % 2 = 0 OFFSET 1 LIMIT 2")
+            .await?
+            .collect()
+            .await?;
+
+        let expected_plan = vec![
+            "+---------------+------------------------------------------------------+",
+            "| plan_type     | plan                                                 |",
+            "+---------------+------------------------------------------------------+",
+            "| logical_plan  | CreateView: \"efg\"                                    |",
+            "|               |   Limit: skip=1, fetch=2                             |",
+            "|               |     Projection: #xyz.column1                         |",
+            "|               |       Filter: #xyz.column2 % Int64(2) = Int64(0)     |",
+            "|               |         TableScan: xyz projection=[column1, column2] |",
```

Comment:
Hi @Dandandan @andygrove . Should the `Limit` and `Filter` be pushed down to the `TableScan`?

Reply:
Yes, when we select from a view, I am expecting these to be pushed down. This will not happen when creating the view though.